### PR TITLE
[convert-branch-deployments-stack] Convert branch deployments guide to Definitions

### PR DESCRIPTION
**Pull request by schrockn**
Summary & Motivation: Converting the branch deployments guide to use Definitions instead of repository

Stacked on https://github.com/dagster-io/dagster/pull/10744

Test Plan: Read

### CODE REVIEW

**Comment by schrockn** [comment]
@benpankow  Added you to this to note where content changes would have to happen once you update the github action for cloud.

**Review by jamiedemaria — COMMENTED** [review]
had one small comment, but looks good to me. Is this going to go along with a similar update to the development to production [guide](https://docs.dagster.io/guides/dagster/transitioning-data-pipelines-from-development-to-production) and [example project](https://github.com/dagster-io/dagster/tree/master/examples/development_to_production)? this guide builds off of the same code example, so it should probably also be updated

**Comment by schrockn** [comment]
@jamiedemaria Should have noted that. That conversion is done in https://github.com/dagster-io/dagster/pull/10744 and the PRs that it depends on.

File: docs/content/guides/dagster/branch_deployments.mdx
Comment (schrockn) [reply docs/content/guides/dagster/branch_deployments.mdx:45]: I actually think we can omit this bullet entirely actually.

File: examples/docs_snippets/docs_snippets/guides/dagster/development_to_production/branch_deployments/clone_prod.yaml
Comment (yuhan) [inline examples/docs_snippets/docs_snippets/guides/dagster/development_to_production/branch_deployments/clone_prod.yaml:26]: is this blocked by @benpankow 's changes in cloud? if so, could we make it clear in the summary as a todo?

**Review by yuhan — APPROVED** [review]
im assuming we are waiting for the cloud action changes before land this right? (either sequencing is fine by me so i'm stamping this one)

**Pull request merged**
